File: no 2/cpp/memory.cpp
class Memory{
	private:
		string frequency;
		string memorySize;
		string supportsCuda;

	public:
		Memory(){
		}

		// set dan get dari private
		void setFrequency(string frequency){
			this->frequency = frequency;
		}
		string getFrequency(){
			return this->frequency;
		}

		void setMemorySize(string memorySize){
			this->memorySize = memorySize;
		}
		string getMemorySize(){
			return this->memorySize;
		}

		void setSupportsCuda(string supportsCuda){
			this->supportsCuda = supportsCuda;
		}
		string getSupportsCuda(){
			return this->supportsCuda;
		}

		~Memory(){
		}


};

File: no 2/php/memory.php
<?php

class Memory extends Hardware{
	private $frequency;
	private $memorySize;
	private $supportsCuda;

	// set dan get 

	function __construct(){
	}

	function setFrequency($frequency){
		$this->frequency = $frequency;
	}
	function getFrequency(){
		return $this->frequency;
	}

	function setMemorySize($memorySize){
		$this->memorySize = $memorySize;
	}
	function getMemorySize(){
		return $this->memorySize;
	}

	function setSupportsCuda($supportsCuda){
		$this->supportsCuda = $supportsCuda;
	}
	function getSupportsCuda(){
		return $this->supportsCuda;
	}

	function __destruct(){

	}

}

?>